client/checkout/blocks/upe-deferred-intent-creation/payment-processor.js

```diff
@@ -0,0 +1,204 @@
+/**
+ * External dependencies
+ */
+import {
+	PaymentElement,
+	useElements,
+	useStripe,
+} from '@stripe/react-stripe-js';
+import {
+	getPaymentMethods,
+	// eslint-disable-next-line import/no-unresolved
+} from '@woocommerce/blocks-registry';
+import { __ } from '@wordpress/i18n';
+import { useEffect, useState } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { usePaymentCompleteHandler } from '../hooks';
+import {
+	getStripeElementOptions,
+	useCustomerData,
+} from 'wcpay/checkout/utils/upe';
+import { getUPEConfig } from 'wcpay/utils/checkout';
+import { validateElements } from 'wcpay/checkout/classic/upe-deferred-intent-creation/payment-processing';
+
+const getBillingDetails = ( billingData ) => {
+	return {
+		name: ( billingData.first_name + ' ' + billingData.last_name ).trim(),
+		email: billingData.email,
+		phone: billingData.phone,
+		address: {
+			city: billingData.city,
+			country: billingData.country,
+
+			line1: billingData.address_1,
+			line2: billingData.address_2,
+			postal_code: billingData.postcode,
+			state: billingData.state,
+		},
+	};
+};
+
+const getFraudPreventionToken = () => {
+	return document
+		.querySelector( '#wcpay-fraud-prevention-token' )
+		?.getAttribute( 'value' );
+};
+
+const PaymentProcessor = ( {
+	api,
+	activePaymentMethod,
+	testingInstructions,
+	eventRegistration: { onPaymentSetup, onCheckoutAfterProcessingWithSuccess },
+	emitResponse,
+	paymentMethodId,
+	upeMethods,
+	errorMessage,
+	shouldSavePayment,
+	fingerprint,
+} ) => {
+	const stripe = useStripe();
+	const elements = useElements();
+	const [ isUPEComplete, setIsUPEComplete ] = useState( false );
+	const [ selectedUPEPaymentType, setSelectedUPEPaymentType ] = useState(
+		''
+	);
+
+	const paymentMethodsConfig = getUPEConfig( 'paymentMethodsConfig' );
+	const isTestMode = getUPEConfig( 'testMode' );
+	const testingInstructionsIfAppropriate = isTestMode
+		? testingInstructions
+		: '';
+	const gatewayConfig = getPaymentMethods()[ upeMethods[ paymentMethodId ] ];
+	const billingData = useCustomerData().billingAddress;
+
+	useEffect(
+		() =>
+			onPaymentSetup( () => {
+				async function handlePaymentProcessing() {
+					if (
+						upeMethods[ paymentMethodId ] !== activePaymentMethod
+					) {
+						return;
+					}
+
+					if ( ! isUPEComplete ) {
+						return {
+							type: 'error',
+							message: __(
+								'Your payment information is incomplete.',
+								'woocommerce-payments'
+							),
+						};
+					}
+
+					if ( errorMessage ) {
+						return {
+							type: 'error',
+							message: errorMessage,
+						};
+					}
+
+					if (
+						gatewayConfig.supports.showSaveOption &&
+						shouldSavePayment &&
+						! paymentMethodsConfig[ selectedUPEPaymentType ]
```

Comment:
```suggestion
						! gatewayConfig.isReusable
```
Minor mention, discussed this in another comment, but I think `selectedUPEPaymentType` will be static in the split UPE, so I think we can use `gatewayConfig` here.

Reply:
Currently, the `isReusable` flag isn't included in `gatewayConfig`. This could be potentially changed [by extending this `supports` array](https://github.com/Automattic/woocommerce-payments/blob/7c5b120eaa82d82c58c0e883a7e93673ef92920a/client/checkout/blocks/upe-split.js#L114-L118). However, taking the setting from `paymentMethodsConfig[ paymentMethodId ].isReusable` seems to be a good candidate as well which also requires fewer changes so I went with this approach. Let me know if this change looks good to you or not 👍

Reply:
Aye aye, cap. I think I misread this one and misunderstood that `gatewayConfig` was returned by blocks' own `getPaymentMethods()` function rather than our own payment methods config. That being said, I'm glad that the redundant `selectedUPEPaymentType` variable has been exterminated, so thanks for making that change!